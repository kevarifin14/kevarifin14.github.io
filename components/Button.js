import { colors } from 'utils';

function Button({ children, onClick, type }) {
  return (
    <>
      <button
        type={type}
        onClick={onClick}
      >
        {children}
      </button>

      <style jsx>{`
        button {
          cursor: pointer;
          padding: 0.5em;
          border-radius: 0;
          font-size: 120%;
          background-color: ${colors.blue};
          color: white;
          border: 0;
        }
      `}</style>
    </>
  );
}

export default Button;