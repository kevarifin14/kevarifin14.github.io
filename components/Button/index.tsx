import { forwardRef, MouseEventHandler, ReactNode } from 'react';

import Spin from 'components/Spin';

export type ButtonProps = {
  type: 'primary' | 'secondary' | 'light',
  children: ReactNode,
  htmlType?: 'submit' | 'button',
  className?: string,
  loading?: boolean,
  disabled?: boolean,
  shape?: 'circle' | 'rounded',
  size?: 'lg',
  onClick?: MouseEventHandler<HTMLButtonElement>,
  block?: boolean,
};

const Button = forwardRef<HTMLButtonElement, ButtonProps>((
  {
    type, children, htmlType = 'button', className, loading, disabled, onClick, block, size,
  },
  ref,
) => {
  const primaryClassName = 'text-white bg-primary hover:bg-primary-dark focus:outline-none border-primary';
  const secondaryClassName = 'text-primary hover:bg-primary-lightest focus:outline-none border-primary-lightest bg-primary-lightest hover:bg-primary-lighter hover:border-primary-lighter';
  const lightClassName = 'text-dark bg-light hover:bg-light-200 focus:outline-none';

  const disabledClassName = 'opacity-50';
  const blockClassName = 'w-full';

  const buttonClassName = [
    'px-4', 'py-2',
    'inline-flex', 'items-center', 'justify-center', 'border',
    'rounded-md', 'shadow-sm', 'focus:outline-none', 'font-medium',
    'focus:ring-2', 'focus:ring-offset-2',
    type === 'primary' ? primaryClassName : '',
    type === 'secondary' ? secondaryClassName : '',
    type === 'light' ? lightClassName : '',
    loading || disabled ? disabledClassName : '',
    block ? blockClassName : '',
    size === 'lg' ? 'px-5 py-3' : '',
    className,
  ].join(' ');

  return (
    <button
      ref={ref}
      type={htmlType}
      className={buttonClassName}
      disabled={disabled || loading}
      onClick={onClick}
    >
      {loading
        && (
          <Spin />
        )}
      {children}
    </button>
  );
});

export default Button;
