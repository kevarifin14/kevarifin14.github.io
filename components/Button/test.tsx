import renderer from 'react-test-renderer';

import Button from '.';

describe('Button', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <Button />,
    );
    expect(tree).toMatchSnapshot();
  });
});
