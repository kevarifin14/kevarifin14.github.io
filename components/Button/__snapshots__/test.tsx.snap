// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<button
  className="px-4 py-2 inline-flex items-center justify-center border rounded-md shadow-sm focus:outline-none font-medium focus:ring-2 focus:ring-offset-2     "
  type="button"
/>
`;
